### DISCUSSION TITLE
Eerste examen IW in het 6e jaar via Dodona 🥳

### DISCUSSION THREAD

**Discussion by DieterPi** [dodona-edu/feedback-and-support#59] [Show and tell]
Gisteren hebben we hier het eerste examen IW voor het 6e jaar via Dodona afgenomen. Alles verliep vlot, maar enkele bedenking/vragen/opmerkingen die ik wil delen.

- We hebben gewerkt met een aparte Dodona cursus (waar ook al toetsen in werden afgenomen tijdens het schooljaar)
   Ik ben wel niet zeker wat de meerwaarde hiervan is. Ik had dit eens gevraagd en op die manier worden oefeningen uit het schooljaar wat afgeschermd, maar 100% waterdicht is dit niet. Via "Profiel" > "Mijn oplossingen" kunnen leerlingen toch steeds aan hun vorige oefeningen? (Ook al staan deze in een andere cursus)
   Is het niet beter om de evaluatie gewoon in de oefeningenreeks af te nemen, maar de vorige 'oefenreeksen' op "verborgen" te plaatsen? Dan kan men totaal niet aan de oefening en oplossing dacht ik?
- Evaluatie werd afgenomen met de Safe Exam Browser. Dit is over het algemeen zeer vlot gegaan. ❤️ Slechts één leerling had wat technische problemen, maar die werden snel opgelost.
   We deelden de geheime link van de reeks via een .seb bestand (met afsluitcode)
- Geëvalueerd met de "anonieme" modus aan. Gewoon top!
- Evaluatiecriteria vind ik moeilijk om vast te leggen. Er zijn vaak verschillende oplossingsmethoden voor de problemen die we hebben bevraagd. (problemen oplossen mbv datastructuren, lijsten, tuples en dictionaries)
   Ik denk dat het Dodona Team of beter nog UGent meer ervaring heeft met een soort van 'rubric' systeem vermoed ik. Kunnen jullie ergens delen wat goed werkt als evaluatiecriteria?

**Comment by jorg-vr** [comment]
Een apparte cursus heeft inderdaad hetzelfde effect als een verborgen reeks binnen de cursus als je die via geheime link gebruikt. Het is dus vooral aan jou als gebruiker om te kiezen hoe je dit liefst structureert.

Ik heb een vraag over Safe Exam Browser: Heb ik het correct voor dat deze jou enkel toelaat om links te volgen die beschikbaar zijn op de originele URL?
In dat geval zouden wij relatief eenvoudig gesloten boek evaluaties moeten kunnen ondersteunen, door op de geheime link van een reeks alle links naar bijvoorbeeld dodona home of het profiel te verwijderen. Want zoals je correct aangeeft, monteel is dit zeker nog niet 100% waterdicht.

De vraag over hoe je een goed rubric opstelt laat ik aan @pdawyndt of andere mensen met meer ervaring hierin.

In elk geval bedankt voor de feedback!

**Reply by DieterPi** [reply]
@jorg-vr Ik denk dat het dan voor mij toch gemakkelijker is om binnen de eigen cursus te werken. Anders krijg ik na een eindje teveel cursussen 🤣. 

Over SEB: Het laat inderdaad toe om een link in te stellen als "startpagina" en waarbij de browser enkel links binnen "het domein" volgt. Maar gezien leerlingen rechts steeds "hun oplossingen" via hun profiel kunnen opvragen kunnen ze dus ook op andere cursussen.

Ervaring heeft ook al geleerd dat de optie "enkel links binnen het eigen domein" toelaten aangevinkt moet staan. (Standaard is dat niet zo) We hebben al leerlingen gevonden die via "informatieve link" naar een wikipedia pagina het spelletje "Six degrees of seperation" speelden. En via via op de wikipedia pagina over chatgpt kwamen en zo uiteindelijk op chatgpt terecht kwamen. 🤣

**Reply by jorg-vr** [reply]
We hadden dit al op onze radar staan, maar ik heb nu ook een issue aangemaakt, zodat we op termijn ook gesloten boek evaluaties beter kunnen ondersteunen: https://github.com/dodona-edu/dodona/issues/6016

Merci voor de info!

**Reply by jonathanvynck** [reply]
@DieterPi Ik vind die setting "enkel links binnen het eigen domein" dus niet terug. Zou je me kunnen vertellen in welk tabblad ik dit terugvind?

**Reply by jorg-vr** [reply]
@jonathanvynck deze setting bestaat nog niet. Momenteel Kan je een reeks op verborgen zetten en dan krijg je een geheime link naar die reeks, maar deze links bevat momenteel wel nog andere referenties, dus studenten kunnen wel nog doorklikken naar de rest van dodona.

Deze discussie heeft geholpen om ons duidelijk te maken waar de nood ligt. We proberen dit binnenkort ook toe te voegen aan Dodona.

**Reply by jorg-vr** [reply]
Ik zie dat je vraag ook al uitgebreid beantwoord is hier: https://github.com/orgs/dodona-edu/discussions/6030 (Ter referentie voor mensen die hier in de toekomst op botsen)

**Reply by pdawyndt** [reply]
@jorg-vr Ik ging ervan uit dat in de vraag van @jonathanvynck "enkel links binnen het eigen domein" een setting is uit de Safe Exam Browser (SEB)

**Reply by DieterPi** [reply]
> @DieterPi Ik vind die setting "enkel links binnen het eigen domein" dus niet terug. Zou je me kunnen vertellen in welk tabblad ik dit terugvind?

Met deze optie "block when directing to a different server". Maar dat werkt natuurlijk enkel voor links die naar 'buiten' dodona sturen.
(Nu zet ik  bij evaluaties eigenlijk ook altijd nieuwe links op "get generally blocked"
![afbeelding](https://github.com/user-attachments/assets/ee768970-4c25-47e9-a77e-3979aa5f7375)

**Reply by jonathanvynck** [reply]
Een aantal leerlingen heeft binnen twee weken een remedieertoets informaticawetenschappen. Ik probeer alles even uit en houd jullie op de hoogte. Een dikke merci alvast voor de hulp!

**Reply by jonathanvynck** [reply]
Ik had toezicht bij het de remedieertoets informaticawetenschappen en de combi SEB met Dodona werkt maar is natuurlijk nog niet perfect.
De exitcode van SEB kan gemakkelijk omzeild worden door de pc handmatig te resetten. Daarna kunnen ze online zaken opzoeken en terug inloggen met SEB.
Omdat ik nu toezicht had bij slechts 10 leerlingen kon ik dit in de gaten houden, maar eens alle vijfdes en zesdes op hetzelfde moment examen hebben, verlies ik het overzicht.
De wifi werkt ook niet overal even goed waardoor leerlingen soms hun opgave niet regelmatig kunnen indienen, maar dat is een probleem bij onze school.

**Reply by DieterPi** [reply]
> Ik had toezicht bij het de remedieertoets informaticawetenschappen en de combi SEB met Dodona werkt maar is natuurlijk nog niet perfect. De exitcode van SEB kan gemakkelijk omzeild worden door de pc handmatig te resetten. Daarna kunnen ze online zaken opzoeken en terug inloggen met SEB. Omdat ik nu toezicht had bij slechts 10 leerlingen kon ik dit in de gaten houden, maar eens alle vijfdes en zesdes op hetzelfde moment examen hebben, verlies ik het overzicht. De wifi werkt ook niet overal even goed waardoor leerlingen soms hun opgave niet regelmatig kunnen indienen, maar dat is een probleem bij onze school.

Akkoord, helemaal perfect is dit niet. Er is zeker nog steeds toezicht nodig. Maar het 'geforceerd' herstarten is toch niet iets wat leerlingen in 1-2-3 uitgevoerd hebben. Ze kunnen ook niet zomeer hun oefening copy pasten in chatgpt of andere tools, enz.. Het geeft mij als toezichter wel veel meer gemoedsrust. Actief toezicht houden blijft nodig, maar ik maak me veel minder zorgen over de rest van de groep als ik eens een vraag van een leerling wil beantwoorden.

We hebben zo'n examen afgenomen in een turnzaal als examenlokaal met meerdere klassen bijeen.

Je kan nog extra beveiliging instellen met een opstartwachtwoord, dan kunnen leerlingen niet in dodona zonder het opstartwachtwoord, maar dan moet de toezichter wel overal het opstartwachtwoord gaan intikken.

**Reply by jonathanvynck** [reply]
Een opstartwachtwoord lijkt me een goed idee.
Deze middag kwam ik toevallig het volgende tegen:
https://www.vvwl.be/veilig-examineren-met-ict-geogebra-excel-python-en-r/
Ik lees zelf nog even na.

Een leerling die met het Kerstexamen heeft gespiekt, scoort nu opnieuw heel goed.
Het is jammer dat ik zo cynisch ben geworden maar ik denk dat hij opnieuw gespiekt heeft, zelfs met SEB.
Tijdens het jaar bakt hij er immers niks van, hij werkt bijzonder traag en nu was hij achter 60 van de 100 minuten klaar, met prima code als resultaat.
Ik ga hem in juni dus heel het examen specifiek in het oog houden...

**Reply by Brechtvsk** [reply]
Is er een mogelijkheid om 1 SB/examen op verschillende deadlines in te plannen voor verschillende gebruikersgroepen? Ik geef het vak IW aan 5 groepen op verschillende tijdstippen en wil dat hun toets verschillende deadline hebben. Hoe lossen jullie dit op? ik heb 5 kopieën gemaakt van mijn toets en elk een verschillende deadline gegeven.

**Reply by pdawyndt** [reply]
@Brechtvsk Dit is ook hoe wij werken: voor elke afzonderlijke examendeadline een afzonderlijke reeks maken in Dodona.

**Reply by DieterPi** [reply]
> Is er een mogelijkheid om 1 SB/examen op verschillende deadlines in te plannen voor verschillende gebruikersgroepen? Ik geef het vak IW aan 5 groepen op verschillende tijdstippen en wil dat hun toets verschillende deadline hebben. Hoe lossen jullie dit op? ik heb 5 kopieën gemaakt van mijn toets en elk een verschillende deadline gegeven.

Ik werk steeds met aparte .seb bestanden per evaluatie. Telkens met ander afsluitwachtwoord.

**Reply by DieterPi** [reply]
> Een opstartwachtwoord lijkt me een goed idee. Deze middag kwam ik toevallig het volgende tegen: https://www.vvwl.be/veilig-examineren-met-ict-geogebra-excel-python-en-r/ Ik lees zelf nog even na.
> 
> Een leerling die met het Kerstexamen heeft gespiekt, scoort nu opnieuw heel goed. Het is jammer dat ik zo cynisch ben geworden maar ik denk dat hij opnieuw gespiekt heeft, zelfs met SEB. Tijdens het jaar bakt hij er immers niks van, hij werkt bijzonder traag en nu was hij achter 60 van de 100 minuten klaar, met prima code als resultaat. Ik ga hem in juni dus heel het examen specifiek in het oog houden...

Als je meer informatie hebt over de specifieke manier waarop de leerling dan 'gespiekt' heeft lijkt me wel interessant om te delen (desnoods via mail). Mocht je dit op een of andere manier weten te achterhalen bedoel ik.

**Reply by jonathanvynck** [reply]
Ik houd jullie op de hoogte.

**Reply by Brechtvsk** [reply]
> > Een opstartwachtwoord lijkt me een goed idee. Deze middag kwam ik toevallig het volgende tegen: https://www.vvwl.be/veilig-examineren-met-ict-geogebra-excel-python-en-r/ Ik lees zelf nog even na.
> > Een leerling die met het Kerstexamen heeft gespiekt, scoort nu opnieuw heel goed. Het is jammer dat ik zo cynisch ben geworden maar ik denk dat hij opnieuw gespiekt heeft, zelfs met SEB. Tijdens het jaar bakt hij er immers niks van, hij werkt bijzonder traag en nu was hij achter 60 van de 100 minuten klaar, met prima code als resultaat. Ik ga hem in juni dus heel het examen specifiek in het oog houden...
> 
> Als je meer informatie hebt over de specifieke manier waarop de leerling dan 'gespiekt' heeft lijkt me wel interessant om te delen (desnoods via mail). Mocht je dit op een of andere manier weten te achterhalen bedoel ik.

Ik heb ook een vermoeden dat er gespiekt is of op de een of andere manier is samengewerkt. Als ik controleer op plagiaat, heb ik 15 leerlingen die 100% match hebben. Het vreemde is dat zij een manier van programmeren gebruiken die we in de les niet hebben gebruikt. Vandaar dat ik vermoed dat ze op de een of andere manier hebben samengewerkt, of de oplossingen hebben gekregen van iemand uit een vorige toets en vanbuiten hebben geleerd. het zou daarom handig zijn mochten we alle SB's onder 1 reeks kunnen stoppen en voor verschillende gebruikers op verschillende momenten inplannen in dodona.

**Reply by DieterPi** [reply]
> Ik heb ook een vermoeden dat er gespiekt is of op de een of andere manier is samengewerkt. Als ik controleer op plagiaat, heb ik 15 leerlingen die 100% match hebben. Het vreemde is dat zij een manier van programmeren gebruiken die we in de les niet hebben gebruikt. Vandaar dat ik vermoed dat ze op de een of andere manier hebben samengewerkt, of de oplossingen hebben gekregen van iemand uit een vorige toets en vanbuiten hebben geleerd. het zou daarom handig zijn mochten we alle SB's onder 1 reeks kunnen stoppen en voor verschillende gebruikers op verschillende momenten inplannen in dodona.

Als je een reeks gebruikt waarbij je de 'geheime link' met de leerlingen deelt (via SEB bvb), dan kunnen leerlingen die de eerste toets gemaakt hebben nadien de opgaven herbekijken. Ze kunnen via hun profiel immers de laatst ingediende oplossingen opvragen. Indien je in de andere klassen dezelfde toetsvragen gesteld hebt, dan waren die toetsvragen inderdaad al beschikbaar...

Dusja, misschien hebben ze thuis de vraag door ChatGPT gehaald en de oplossing uit het hoofd geleerd. Ik maak steeds verschillende oefenreeksen per klas. (zeker als er een dag tussenzit)

**Reply by heremansgeert** [reply]
Wij hebben vorig jaar leerlingen betrapt op spieken ondanks het gebruik van de SEB. De leerlingen hadden een virtuele machine geïnstalleerd waarop ze de SEB startte. De toezichthoudende leerkracht kan dat niet eenvoudig zien dat dit in een VM gebeurde.

SEB heeft een beveiliging tegen het draaien op VM's maar deze is erg [eenvoudig uit te schakelen](https://github.com/nxvvvv/safe-exam-browser-bypass).  

Verder zijn er ook gekraakte SEB versies in omloop. Wij gebruiken schermmonotoring software om dit tegen te gaan.

**Reply by heremansgeert** [reply]
Verder wil ik ook nog graag aangeven dat wij in onze vakgroep ook worstelen met het opstellen van goede en objectieve evaluatiecriteria voor programmeeroefeningen. Ik heb het gevoel dat eenzelfde oefening door 2 verschillende leerkrachten wel eens heel anders beoordeeld zou kunnen worden.

Mochten de mensen van UGent of anderen evaringen daarover kunnen en willen delen dan verneem ik dat ook graag :-)

**Reply by IngeVerbeek** [reply]
> Ik zie dat je vraag ook al uitgebreid beantwoord is hier: https://github.com/orgs/dodona-edu/discussions/6030 (Ter referentie voor mensen die hier in de toekomst op botsen)

Bij de start van het nieuwe schooljaar bots ik hier (weer) op. Vorig jaar hadden we een aparte cursus voor toetsen en examens, maar ik begrijp dat dat niet zinvol is omdat dit hetzelfde effect is als een verborgen link. Ik wou doorklikken naar die discussie, maar dat geeft een error.

Staat "gesloten boek evaluaties" op jullie (kortetermijn)planning?

**Reply by bmesuere** [reply]
@IngeVerbeek een examenmodus staat met prioriteit op onze planning eenmaal de drukte van de start van het schooljaar achter de rug is.

**Reply by IngeVerbeek** [reply]
@bmesuere Hebben jullie een idee of die examenmodus al bruikbaar zal zijn bij de kerstexamens begin december?

Ik stoot momenteel op (onder andere) de volgende problemen bij evaluaties:

- Sommige leerlingen zijn vroeger klaar en mogen volgens ons examenreglement dan afsluiten en vertrekken. Ze kunnen dan buiten het examenlokaal overleggen (of hulp vragen) en nog een betere oplossing indienen. Ik kan natuurlijk hun vertrekuur noteren en handmatig controleren of ze niets meer indienen daarna, maar dit is omslachtig. Ik zou het handig vinden als ze ergens "definitief indienen" moeten klikken of zo, wat ik dan kan controleren voor ze buiten gaan.

- Vorig jaar stelde ik vast dat een leerling (weliswaar per ongeluk) op het account van een andere leerling aan het werken was, omdat ze even voordien nog samen geoefend hadden. Er waren dus twee leerlingen tegelijk oplossingen aan het indienen op één account. Dit is natuurlijk een vrijgeleide voor plagiaat: de leerling logt in op school tijdens het examen en doet alsof hij/zij werkt, en tegelijk legt thuis iemand anders het examen af en dient goede oplossingen in. Kan dit vermeden worden in de examenmodus? 
We proberen dit nu tegen te gaan door de opgave op papier uit te delen en in de description alleen een afbeelding te zetten die naar de opgave verwijst, maar dat is niet waterdicht natuurlijk. 

- We werken op school met https://classroom.cloud/, daarmee kan ik toegang tot bepaalde applicaties en/of websites blokkeren. Ik werkte al met evaluaties via "verborgen reeks" die alleen toegankelijk is met de link. Die link moet ik natuurlijk versturen naar de leerlingen via Smartschool. Als leerlingen per ongeluk afsluiten, moeten ze terug via smartschool naar de geheime link. Dat is vervelend, omdat ik de toegang tot smartschool afsluit nadat ze zijn ingelogd (anders kunnen ze aan de cursus-pdf, kunnen ze berichten sturen via smartschool, ...). Het zou handiger zijn als ik op Dodona zelf kan aanvinken welke leerlingen wel/niet toegang krijgen tot de verborgen reeks, zodat de leerlingen alleen op dat het Dodona-platform kunnen blijven.

**Reply by bmesuere** [reply]
Dag @IngeVerbeek, we hebben ondertussen al de analysefase van de examenmodus achter de rug en beginnen nu aan de implementatie. Ik kan niet beloven dat deze voor de examenperiode voor iedereen beschikbaar zal zijn. Mogelijk werken we voor de examenperiode van december nog met enkele pilootgebruikers om zeker te zijn dat alles goed zit. Wat betreft de problemen die je opsomt: ik denk dat we voor de meeste een oplossing of minstens grotere verbetering zullen kunnen kiezen.

- definitief indienen: staat gepland als basisfeature van de examenmodus.
- dubbel inloggen verhinderen/sessie koppelen aan ip-adres: we kunnen dit niet 100% verhinderen door de manier waarop het internet werkt, maar we hebben wel een plan om dit risico significant te beperken.
- toegang tot de examenreeks zullen we ook herwerken dus dit probleem zal grotendeels opgelost worden. Er staat ook integratie met safe exam browser/school year/netsupport op de planning, maar dit zal waarschijnlijk niet in de eerste release zijn. Ik kende classroom.cloud niet, maar dit lijkt te werken op basis van netsupport. We nemen het mee.
